pkg/stanza/fileconsumer/roller_other.go

```diff
@@ -30,15 +30,18 @@ OUTER:
 			if newReader.Fingerprint.StartsWith(oldReader.Fingerprint) {
 				continue OUTER
 			}
-			if oldReader.FileName == newReader.FileName {
-				// At this point, we know that the file has been rotated. However, we do not know
-				// if it was moved or truncated. If truncated, then both handles point to the same
-				// file, in which case we should only read from it using the new reader. We can use
-				// the validateFingerprint method to establish that the file has not been truncated.
-				if !oldReader.ValidateFingerprint() {
-					continue OUTER
-				}
+			if !newReader.NameEquals(oldReader) {
```

Comment:
Is this a change of behavior?

Reply:
I don't think so. The equal check is moved into the function.

The inversion of the condition is met with a complimentary inversion of how we react to it.

Reply:
You're right, this was a change. It should only continue the inner loop if the file names do not match. I opened https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27454 to make the change separately. I'll plan on rebasing this PR.